### PR TITLE
Fix PDF filename variables for PDF service tasks without layout-set (auto PDF)

### PR DESCRIPTION
**Pull request by bjorntore**
Fix PDF filename variables for PDF service tasks without layout-set (auto PDF)

## Description
Returning a LayoutEvaluatorState even if no layout-set is found for the task. LayoutEvaluatorState already has LayoutState as a nullable, so I think it's fine? @ivarne 

## Related Issue(s)
- [#17522](https://github.com/Altinn/altinn-studio/issues/17522)

## Verification
- [x] **Your** code builds clean without any errors or warnings
- [x] Manual testing done (required)
- [x] Relevant automated test added (if you find this hard, leave it and we'll help out)
- [x] All tests run green

## Documentation
- [ ] User documentation is updated with a separate linked PR in [altinn-studio-docs.](https://github.com/Altinn/altinn-studio-docs) (if applicable)

## Summary by CodeRabbit

* **Bug Fixes**
  * Enhanced error detection and reporting with more descriptive messages when translating text in specific configuration scenarios.

* **Refactor**
  * Refactored internal logic to reduce code duplication and improve consistency across file naming and text translation operations for better maintainability.

* **Tests**
  * Added comprehensive new test cases covering translation service edge cases and configuration scenarios to improve test coverage and reliability.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

**Review by ivarne — APPROVED** [review]
This looks correct.

The better approach would be to take `IInstanceDataAccessor` in the `ComponentContext` instead, and just call `GetLayoutEvaluatorState` when it is actually needed.

I want to get rid of `LayoutEvaluatorState` at some point, but this has way less impact

File: src/Altinn.App.Core/Internal/Data/InstanceDataUnitOfWork.cs [hunk 1]
```diff
@@ -189,10 +189,6 @@ public IInstanceDataAccessor GetPreviousDataAccessor()
 
         // Could use a double lock here, but a deadlock is more problematic than creating the state twice
         var layouts = _appResources.GetLayoutModelForTask(TaskId);
-        if (layouts is null)
```
Comment (bjorntore) [inline src/Altinn.App.Core/Internal/Data/InstanceDataUnitOfWork.cs:192]: This is the main change @ivarne

File: test/Altinn.App.Core.Tests/Internal/Texts/TranslationServiceInstanceTests.cs [hunk 2]
```diff
@@ -77,4 +78,89 @@
...
+                Id = "testResource",
+                Value = "Value: {0}",
+                Variables = [new TextResourceVariable() { DataSource = "dataModel.SkjemaModel", Key = "Input1" }],
+            }
+        );
+
+        fixture.TryAddCommonServices();
+
+        // Override the mock to return null instead of throwing for GetLayoutModelForTask
+        fixture.AppResourcesMock.Setup(a => a.GetLayoutModelForTask(It.IsAny<string>())).Returns((LayoutModel?)null);
```
[the remark this answers is not in the document]
Comment (ivarne) [reply test/Altinn.App.Core.Tests/Internal/Texts/TranslationServiceInstanceTests.cs]: This is wrong. I have tried to put `<LayoutModel?>` somewhere instead of the cast, but did not manage to get it to work with Moq.

**Review by ivarne — APPROVED** [review]
Good catch by Coderabit

**Review by olemartinorg — APPROVED** [review]
I didn't read the code, but ran the e2e tests from frontend, and they're working now. 🥳
